src/utils/publishToNpm.js

```diff
@@ -63,18 +63,57 @@ async function publishToNpm({
   if (await allowNpmPublish(version)) {
     await execWithOutput('npm', ['pack', '--dry-run'])
 
-    if (opticToken) {
-      const packageInfo = await getLocalInfo()
-      const otp = await execWithOutput('curl', [
-        '-s',
-        '-d',
-        JSON.stringify({ packageInfo: { version, name: packageInfo?.name } }),
-        '-H',
-        'Content-Type: application/json',
-        '-X',
-        'POST',
-        `${opticUrl}${opticToken}`,
-      ])
+    let otp
+
+    if (opticToken || ngrokApiKey) {
+      if (opticToken) {
+        const packageInfo = await getLocalInfo()
+        
+        // push notification via optic mobile app
+        otp = await execWithOutput('curl', [
+          '-s',
+          '-d',
+          JSON.stringify({ packageInfo: { version, name: packageInfo?.name } }),
+          '-H',
+          'Content-Type: application/json',
+          '-X',
+          'POST',
+          `${opticUrl}${opticToken}`,
+        ])
+      } else if (ngrokApiKey) {
+        // 1. run a fastify app which exposes a simple web UI with a text input to get the OTP
+        // 2. expose the app via ngrok
+        // 3. print the url to the app on the workflow execution output
+        // 4. wait for some time or a timeout
+        // 5. if we get the token from the user, continue with the execution, otherwise timeout
+        const app = fastify()
+
+        app.get('/', () => {
+          // serve static page with text input which, when submitted, POSTs the OTP to /otp url
+        })
+
+        let otpPromiseResolve
+        const otpPromise = new Promise(resolve => { otpPromiseResolve = resolve })
+
+        app.post('/otp', async req => {
+          const otp = req.body.otp
+
+          otpPromiseResolve(otp)
+        })
+
+        app.listen({ host: ..., port: ...})
+
+        // expose the fastify app via nkgrok and get the ngrok app url, see https://github.com/ngrok/ngrok-javascript
+        const ngrokUrl = ...
```

Comment:
we said that it's probably easier to use localtunnel instead of ngrok, but the api is very similar anyway